frontend/src/pages/EventCreationPage/components/EventCreationForm/index.tsx

```diff
@@ -29,9 +29,29 @@ interface InitialValuesType {
   hosts: OfficerNameData[];
 }
 
+function findNearestHour(date, isStartDate) {
+  let hour = date.getHours();
+  let startMinutes = date.getMinutes();
+  if (startMinutes === 0) {
+    startMinutes = '00';
+  } else if (startMinutes <= 30) {
+    startMinutes = '30';
+  } else if (startMinutes > 30) {
+    startMinutes = '00';
+    hour += 1;
+  }
+
+  date.setMinutes(startMinutes);
+  date.setHours(hour);
+  if (!isStartDate) {
```

Comment:
since the second `isStartDate` parameter only adds an hour to the endDate, `findNearestHour` would be more flexible if we removed the isStartDate parameter and instead did the hour adding outside the function since that preserves `findNearestHour` as doing that thing only, and allows us to reuse it elsewhere outside of the event start end date contexts.

see https://en.wikipedia.org/wiki/Inversion_of_control if you're interested in more details! :)